providers/flagd/go.mod

```diff
@@ -1,8 +1,6 @@
 module github.com/open-feature/go-sdk-contrib/providers/flagd
 
-go 1.21
```

Comment:
Do you have any idea why this keeps being added?

Reply:
This comes from Renovate and there's an option to control the version - https://docs.renovatebot.com/golang/#go-binary-version 

However, given this is a mono repo with multiple providers, we cannot enforce this as each provider may support their desired go version. 

As maintainers however, we should allow our CI to run with the latest version of Go as there's a backward compatibility promise.